### ISSUE TITLE
MVM marked as CREATED despite ongoing network reconciliation errors

### ISSUE THREAD

**Issue by Callisto13** [weaveworks/flintlock#313]
**What happened:**

- Created MicroVM
- Saw in logs:
   ```
   ERRO[16896] failed executing plan                         controller=microvm execution_id=01FPDAAFTKMKF1AMZ2XWFJFA0R execution_time=2.282275ms num_steps=1 plan_name=microvm_create_update
   INFO[16896] [5/10] reconciliation failed, rescheduled for next attempt at 2021-12-08 15:26:00 +0000 GMT  action=reconcile controller=microvm vmid=default/mvm-test-control-plane-9855w
   DEBU[2021-12-08T15:24:20.694283264Z] (*service).Write started                      ref=works.weave.flintlockd/microvm/default/mvm-test-control-plane-9855w
   ERRO[16896] failed to reconcile vmid default/mvm-test-control-plane-9855w: executing plan: executing plan steps: executing steps: executing step network_iface_create: creating network interface: creating interface flvtapdbf672d using netlink: cannot assign requested address  controller=microvm
   ```
- When Getting the mvm via the api, the status is reported as CREATED
- As this is our highest state, CAPMVM sees it as "running" and ready

**What did you expect to happen:**
The MVM should not have been in a created state.

**How to reproduce it:**
I think the cause of the error itself is because I deleted my tap interface when I was messing around with bridges, but you could probably just get this to happen by making something in `IfaceCreate` in `infrastructure/network/network_service.go` fail (like `if true { return err }` or smth).

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- flintlock version:
- containerd version:
- OS (e.g. from `/etc/os-release`):

**Comment by Callisto13** [comment]
We seem to set the state to Created (if it was previously Pending) right at the start of the Create plan :thinking-face:

**Comment by Callisto13** [comment]
more weirdness and issues found:

1. When I hardcoded a fail in the IfaceCreate method, it failed the reconciliation twice... then on the third decided it was okay 😂 , and the "create" went through. No sign of a firecracker process at the end of this. Seems the ShouldDo of the network step is a bit off?
2. Watching closely, we don't actually set any status at all until that Create is added before steps are executed.

**Comment by Callisto13** [comment]
Okay so that was an adventure.

So it turned out that the GRPC client we use to poke flintlock for some reason does not like the enum values returned in our responses. When I wrote my own client, everything was fine:

```bash
 ./hack/scripts/send.sh -m GetMicroVM | jq .[].status.state
null  # does not like the pending enum for "reasons" and completely omits that field

go run main.go # this is doing the same filtering as above
PENDING # totally fine 🤷‍♀️ 
```

The good news is that things are fine on our end. Which leaves the following issues:
- We need to stop hardcoding Created before we have even created anything
- The Interface steps need a better ShouldDo check

**Comment by Callisto13** [comment]
> When I hardcoded a fail in the IfaceCreate method, it failed the reconciliation twice... then on the third decided it was okay 😂 , and the "create" went through. No sign of a firecracker process at the end of this. Seems the ShouldDo of the network step is a bit off?

Actually can't get this to happen again, so think this one is done
